### DISCUSSION TITLE
Make the id option consistent between spawn and invoke API

### DISCUSSION THREAD

**Discussion by rlaffers** [statelyai/xstate#2069] [Ideas]
To assign an ID to an invoked service, one uses the `id` option. E.g.:
```javascript
invoke: {
  src: someMachine,
  id: 'machine1',
}
```
To assign an ID to a spawned actor, one has to use the `name` option. E.g.:
```javascript
entry: assign(
  actorRef: () => spawn(someMachine, { name: 'machine2' })
)
```
Both spawned actors, and invoked services are listed in the `service.children` map side by side, keyed by their id/name.

Is there a deeper reason for the discrepancy in how we assign IDs to them? If not, it would make sense to specify IDs in both cases with an `id` option. E.g. (proposed):
```javascript
entry: assign(
  actorRef: () => spawn(someMachine, { id: 'machine2' })
)
```

**Comment by Andarist** [comment]
Could u create this as an issue? We could then manage this in a better way.

**Reply by rlaffers** [reply]
https://github.com/davidkpiano/xstate/issues/2070
